src/rpc/rawtransaction.cpp

```diff
@@ -759,22 +763,44 @@ static UniValue combinerawtransaction(const JSONRPCRequest& request)
     return EncodeHexTx(mergedTx);
 }
 
-UniValue SignTransaction(CMutableTransaction& mtx, const UniValue& prevTxsUnival, CBasicKeyStore *keystore, bool is_temp_keystore, const UniValue& hashType)
+// Wrapper to help add a single coin to CCoinsViewCache.
+class CoinFill : private CCoinsView
+{
+public:
+    CoinFill(CCoinsViewCache& cache, const COutPoint &output, Coin&& coin, CCoinsView &backend) : m_cache(cache), m_output(output), m_coin(std::move(coin)), m_backend(backend) {
+        m_cache.SetBackend(*this);
+        m_cache.AccessCoin(output);
+    }
+    ~CoinFill() {
+        m_cache.SetBackend(m_backend);
+    }
+
+    private:
+    bool GetCoin(const COutPoint &output, Coin &coin) const override {
+        assert(output == m_output);
+        coin = std::move(m_coin);
+        return true;
+    }
+    CCoinsViewCache& m_cache;
+    const COutPoint& m_output;
+    Coin&& m_coin;
+    CCoinsView& m_backend;
+};
+
+UniValue SignTransaction(interfaces::Chain& chain, CMutableTransaction& mtx, const UniValue& prevTxsUnival, CBasicKeyStore *keystore, bool is_temp_keystore, const UniValue& hashType)
 {
     // Fetch previous transactions (inputs):
     CCoinsView viewDummy;
     CCoinsViewCache view(&viewDummy);
     {
-        LOCK2(cs_main, mempool.cs);
-        CCoinsViewCache &viewChain = *pcoinsTip;
-        CCoinsViewMemPool viewMempool(&viewChain, mempool);
-        view.SetBackend(viewMempool); // temporarily switch cache backend to db+mempool view
-
+        std::vector<COutPoint> outputs;
         for (const CTxIn& txin : mtx.vin) {
-            view.AccessCoin(txin.prevout); // Load entries from viewChain into view; can fail.
+            outputs.emplace_back(txin.prevout);
+        }
+        std::vector<Coin> coins = chain.findCoins(outputs);
+        for (int i = 0; i < coins.size(); ++i) {
```

Comment:
perhaps change `int` to `size_t` to prevent comparison of integers with different sign warning?

Reply:
> perhaps change int to size_t to prevent comparison of integers with different sign warning?

Will fix, but this change is part of the base PR #10973 (249bf5006184f81d77d40ee0ede0924c628bf33e), so it would be better to comment on that PR.